sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -203,7 +203,23 @@ abstract class RapidsShuffleInternalManagerBase(conf: SparkConf, isDriver: Boole
   def getServerId: BlockManagerId = server.fold(blockManager.blockManagerId)(_.getId)
 
   override def addPeer(peer: BlockManagerId): Unit = {
-    transport.foreach(_.connect(peer))
+    transport.foreach { t =>
+      try {
+        t.connect(peer)
+      } catch {
+        case ex: Exception =>
+          // We ignore the exception after logging in this instance because
+          // we may have a peer that doesn't exist anymore by the time `addPeer` is invoked
+          // due to a heartbeat response from the driver, or the peer may have a temporary network
+          // issue.
+          //
+          // This is safe because `addPeer` is only invoked due to a heartbeat that is used to
+          // opportunistically hide cost of initializing transport connections. The transport
+          // will re-try if it must fetch from this executor at a later time, in that case
+          // a connection failure causes the tasks to fail.
+          logError(s"Unable to connect to peer $peer, ignoring!", ex)
```

Comment:
Should this be a warning rather than an error?  Error implies something is not going to work whereas a warning implies something odd happened but the situation can be recovered.

Reply:
Fixed in: https://github.com/NVIDIA/spark-rapids/pull/2587/commits/87f14f63bf9e4c24729e7ffb6e2b1545d475382a